### ISSUE TITLE
XL:  ReedSolomon joining decoded blocks failed with io: read/write on closed pipe.  (May 14th)

### ISSUE THREAD

**Issue by hackintoshrao** [minio/minio#1618]
This error is produced only high concurrent execution of [exec-concurrent.go](https://gist.github.com/hackintoshrao/6f821cc7185e39cce3733c138390fc29) , that is when concurrency level is set to around 100, even at 50 this issue is not produced. 

Here is the server error log 

`ReedSolomon joining decoded blocks failed with io: read/write on closed pipe  path=a1r6onxx9gcv2dawllnvagxkub0xcf-copy volume=jvxueohu1hxuierut1jo0k60vmn63r-copy` 

Once this happens the tests time out waiting for response form server, Here is the concurrent.log (play/sanctity/concurrent1.log). 

In the concurrent log you can see the point where it was blocked before the testing framework decided to time them out. Not all goroutines failed, around less half of them failed. They were mostly blocked on GetObject when they were abruptly stopped.

**Comment by harshavardhana** [comment]
This is interesting, io.Pipe() has closed before while the reader was still trying to decode.

**Comment by hackintoshrao** [comment]
@harshavardhana  : Able to consistently reproduce it with concurrency levels great than 100.

**Comment by harshavardhana** [comment]
> @harshavardhana : Able to consistently reproduce it with concurrency levels great than 100.

Cool, let me know where is the setup, i can take a look.

**Comment by hackintoshrao** [comment]
@harshavardhana : Its on amazon instance.

**Comment by hackintoshrao** [comment]
All the failed Goroutines were blocked on getObject. The get object stops responding after a while. 
It is clear form the log 
`cat concurrent.log | grep api-get-object.go` 

```
  /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d2
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:92 +0x3c5
        /home/ec2-user/mygo/src/github.com/minio/minio-go/api-get-object.go:59 +0x4d
```

**Comment by harshavardhana** [comment]
Is this still valid?

**Comment by harshavardhana** [comment]
Closing this as this was fixed by #1629
